internal/app/machined/pkg/controllers/runtime/kernel_param_spec.go

```diff
@@ -73,6 +74,12 @@ func (ctrl *KernelParamSpecController) Run(ctx context.Context, r controller.Run
 		case <-ctx.Done():
 			return nil
 		case <-r.EventCh():
+			ksppParams := map[string]struct{}{}
+
+			for _, param := range kspp.GetKernelParams() {
+				ksppParams[param.Key] = struct{}{}
```

Comment:
I would guess it's not the right key here? as KSPP params have sysctl names as keys?

Reply:
yeah... I don't like the code still, trying to make it more reliable and straightforward.

Reply:
finally it looks ok.